apps/meteor/tests/e2e/team-management.spec.ts

```diff
@@ -105,7 +186,119 @@ test.describe.serial('teams-management', () => {
+		await expect(page.getByRole('menu', { exact: true }).getByRole('menuitem', { name: 'Remove from team' })).toBeVisible();
+	});
+
+	test('should not allow deleting a targetGroup from targetTeam if the group owner does not have the delete-team-group permission', async ({
+		page,
+		api,
+	}) => {
+		expect((await api.post('/permissions.update', { permissions: [{ _id: 'delete-team-group', roles: ['moderator'] }] })).status()).toBe(
+			200,
+		);
+
+		await poHomeTeam.sidenav.openChat(targetTeam);
+		await poHomeTeam.tabs.btnChannels.click();
+		await poHomeTeam.tabs.channels.openChannelOptionMoreActions(targetGroupNameInTeam);
+		await expect(page.getByRole('menu', { exact: true }).getByRole('menuitem', { name: 'Delete' })).not.toBeVisible();
+	});
+
+	test('should allow deleting a targetGroup from targetTeam if the group owner also has the delete-team-group permission', async ({
+		page,
+		api,
+	}) => {
+		expect((await api.post('/permissions.update', { permissions: [{ _id: 'delete-team-group', roles: ['owner'] }] })).status()).toBe(200);
+
+		await poHomeTeam.sidenav.openChat(targetTeam);
+		await poHomeTeam.tabs.btnChannels.click();
+		await poHomeTeam.tabs.channels.openChannelOptionMoreActions(targetGroupNameInTeam);
+		await expect(page.getByRole('menu', { exact: true }).getByRole('menuitem', { name: 'Delete' })).toBeVisible();
+		await page.getByRole('menu', { exact: true }).getByRole('menuitem', { name: 'Delete' }).click();
+		await poHomeTeam.tabs.channels.confirmDeleteRoom();
+
+		await poHomeTeam.sidenav.openChat(targetTeam);
+		await poHomeTeam.tabs.btnChannels.click();
+		await expect(poHomeTeam.tabs.channels.channelsList).not.toContainText(targetGroupNameInTeam);
+	});
+
+	test('should not allow removing a targetChannel from targetTeam if user does not have the remove-team-channel permission', async ({
+		page,
+		api,
+	}) => {
+		expect((await api.post('/permissions.update', { permissions: [{ _id: 'remove-team-channel', roles: ['moderator'] }] })).status()).toBe(
+			200,
+		);
+
+		await poHomeTeam.sidenav.openChat(targetTeam);
+		await poHomeTeam.tabs.btnChannels.click();
+		await poHomeTeam.tabs.channels.openChannelOptionMoreActions(targetChannelNameInTeam);
+		await expect(page.getByRole('menu', { exact: true }).getByRole('menuitem', { name: 'Remove from team' })).not.toBeVisible();
+	});
+
+	test('should allow removing a targetChannel from targetTeam if user has the remove-team-channel permission', async ({ page, api }) => {
+		expect((await api.post('/permissions.update', { permissions: [{ _id: 'remove-team-channel', roles: ['owner'] }] })).status()).toBe(200);
+
+		await poHomeTeam.sidenav.openChat(targetTeam);
+		await poHomeTeam.tabs.btnChannels.click();
+		await poHomeTeam.tabs.channels.openChannelOptionMoreActions(targetChannelNameInTeam);
+		await expect(page.getByRole('menu', { exact: true }).getByRole('menuitem', { name: 'Remove from team' })).toBeVisible();
```

Comment:
Since we're already with the page open, maybe test if the action works too?

Reply:
Done! I had to re-add the room to the team in the next test to keep them working, though